src/Zynga/Framework/PgData/V1/PgModel/Cache.hh

```diff
@@ -71,21 +75,27 @@ class Cache implements CacheInterface {
     }
 
   }
-  
+
   public function clearResultSetCache<TModelClass as PgRowInterface>(
     classname<TModelClass> $model,
     PgWhereClauseInterface $where,
   ): bool {
     try {
+
       $pgModel = $this->pgModel();
 
       // Create a cachedRs out of the result set that was given.
       $cachedRs = $pgModel->reader()->createCachedResultSet($model, $where);
 
       $cache = $this->getResultSetCache();
 
-      if($cache->lock($cachedRs)) {
-        return $cache->delete($cachedRs, true);
+      if ($cache->lock($cachedRs)) {
+        $result = $cache->delete($cachedRs, false);
+        if ($result && $pgModel->allowWriterOverride()) {
+          $this->setWriterOverrideForWhereClause($model, $where);
```

Comment:
can we log an error when this fails to write just so we have logs on IF and when it happens